### PR TITLE
Update HAProxy deploy operating system introduction.

### PR DESCRIPTION
**Pull request by superlzs0476**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by superlzs0476** [comment]
@qiancai  PTAL

**Review by qiancai — APPROVED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -42,19 +42,20 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
...
+| Linux 操作系统平台       | 版本         |
+| :----------------------- | :----------: |
+| Red Hat Enterprise Linux | 7 或者 8   |
+| CentOS                   | 7 或者 8   |
+| Oracle Enterprise Linux  | 7 或者 8   |
+| Ubuntu LTS               | 20.04 或者 18.04 |
+
+> **注意：**
+>
+> - 其他操作系统支持情况，详见 [HAProxy 官方文档](https://www.haproxy.com/documentation/hapee/latest/getting-started/installation/linux/)
```
Comment (qiancai) [inline best-practices/haproxy-best-practices.md:58]:
```suggestion
> - 其他操作系统支持情况，详见 [HAProxy 官方文档](https://www.haproxy.com/documentation/hapee/latest/getting-started/installation/linux/)。
```

**Review by superlzs0476 — COMMENTED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -41,19 +41,20 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
...
+| Linux 操作系统平台       | 版本         |
+| :----------------------- | :----------: |
+| Red Hat Enterprise Linux | 7 或者 8   |
+| CentOS                   | 7 或者 8   |
+| Oracle Enterprise Linux  | 7 或者 8   |
+| Ubuntu LTS               | 20.04 或者 18.04 |
+
+> **注意：**
+>
+> - 其他操作系统支持情况，详见 [HAProxy 官方文档](https://www.haproxy.com/documentation/hapee/latest/getting-started/installation/linux/)。
```
Comment (qiancai) [inline best-practices/haproxy-best-practices.md:57]:
```suggestion
> - 其他操作系统支持情况，详见 [HAProxy 文档](https://github.com/haproxy/haproxy/blob/master/INSTALL)。
```

File: best-practices/haproxy-best-practices.md [hunk 3]
```diff
@@ -41,19 +41,20 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
...
-| 操作系统版本               | 架构                                       |
-|:-------------------------|:------------------------------------------|
-| Linux 2.4                | x86、x86_64、Alpha、SPARC、MIPS 和 PA-RISC  |
-| Linux 2.6 或 3.x         | x86、x86_64、ARM、SPARC 和 PPC64            |
-| Solaris 8 或 9           | UltraSPARC II 和 UltraSPARC III            |
-| Solaris 10               | Opteron 和 UltraSPARC                      |
-| FreeBSD 4.10 ~ 10        | x86                                        |
-| OpenBSD 3.1 及以上版本     | i386、AMD64、macppc、Alpha 和 SPARC64       |
-| AIX 5.1 ~ 5.3            | Power™                                     |
+| Linux 操作系统平台       | 版本         |
```
Comment (Chad20N13) [inline best-practices/haproxy-best-practices.md:48]: | OS platform      | Version                    |
|----------------|--------------------|
| Linux kernel      | 2.6.28 and above   |
| Solaris               | 10 and above          | 
| FreeBSD            | 10 and above         | 
| DragonFlyBSD  | 4.3 and above        | 
| NetBSD             | 8 and above            |
| Mac OS/X          | N/A                           | 
| OpenBSD          | 6.3 and above          | 
| AIX                     | 5.1 and 5.2 and 7.2  | 
| Cygwin              | N/A                            | 
| Haiku                 | N/A                            |

> **注意：**
>
> - 其他操作系统支持情况，详见 [HAProxy 文档](https://github.com/haproxy/haproxy/blob/master/INSTALL)。

File: best-practices/haproxy-best-practices.md [hunk 4]
```diff
@@ -41,19 +41,20 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
...
+| :----------------------- | :----------: |
```
Comment (qiancai) [inline best-practices/haproxy-best-practices.md:49]:
```suggestion
| :----------------------- | :----------- |
```

**Review by Chad20N13 — APPROVED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 5]
```diff
@@ -41,19 +41,20 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
...
-| Solaris 10               | Opteron 和 UltraSPARC                      |
-| FreeBSD 4.10 ~ 10        | x86                                        |
-| OpenBSD 3.1 及以上版本     | i386、AMD64、macppc、Alpha 和 SPARC64       |
-| AIX 5.1 ~ 5.3            | Power™                                     |
+| Linux 操作系统平台       | 版本         |
+| :----------------------- | :----------- |
+| Red Hat Enterprise Linux | 7 或者 8   |
+| CentOS                   | 7 或者 8   |
+| Oracle Enterprise Linux  | 7 或者 8   |
+| Ubuntu LTS               | 20.04 或者 18.04 |
```
Comment (qiancai) [inline best-practices/haproxy-best-practices.md:53]:
```suggestion
| Ubuntu LTS               | 18.04 或者以上版本 |
```

**Review by TomShawn — APPROVED** [review]

**Pull request merged**